### PR TITLE
Rework editing of permissions

### LINKED ISSUE #245: ui: Rework editing of permissions
## Current situation
When editing permissions, each click performs an API action and grants/revokes permissions. This is not ideal for some reasons: 
* If an error was made, it is written permanently to the chain
* The scenario where a user grants permissions, but cannot revoke them is possible - which combined with the first issue can cause problems
* There is no way to restore the original state during edit (i.e. no "cancel" button)

## Proposed solution
Similar to the drawer in the [batch edit function for workflowitems](https://github.com/openkfw/TruBudget/blob/master/doc/wiki/User-Guide/Projects/Workflowitem.md#update-multiple-workflowitems), the permissions should be changed in the permissions window without any action on API level and only after click of "apply" the changes are persisted to the chain. 
Additionally, add a "cancel" button that simply cancels the process and takes the user back to the previous screen. 
This should be done for global permissions as well as permissions for projects/subprojects/workflowitems.

### PR DESCRIPTION
**Pull request by mathiashoeld**
# Changes
The user now has to submit the changes that were made to the permissions of a project, subproject or workflow item. 

Closes #245 

# How to test
Change the permissions for e.g. a project, click submit and the permission should have changed. Try the same and hit cancel instead and the permissions should be the same as before opening the dialog.

### CODE REVIEW

File: frontend/src/pages/Overview/reducer.js [hunk 1]
```diff
@@ -76,7 +79,8 @@ export default function overviewReducer(state = defaultState, action) {
       return state.merge({
         idForPermissions: defaultState.get("id"),
         permissionDialogShown: false,
```
Comment (Stezido) [inline frontend/src/pages/Overview/reducer.js]: why not use the defaultState here?

File: frontend/src/pages/SubProjects/reducer.js [hunk 2]
```diff
@@ -90,6 +93,7 @@ export default function detailviewReducer(state = defaultState, action) {
     case SHOW_SUBPROJECT_PERMISSIONS:
       return state.merge({
         permissions: fromJS({}),
```
Comment (Stezido) [inline frontend/src/pages/SubProjects/reducer.js]: Why not use the defaultState here?

File: frontend/src/pages/Users/Dialog.js [hunk 3]
```diff
@@ -86,58 +94,42 @@ const Dialog = props => {
       };
       break;
     case "editUserPermissions":
+      console.log(temporaryGlobalPermissions);
```
Comment (Stezido) [inline frontend/src/pages/Users/Dialog.js]: remove this

File: frontend/src/pages/Users/reducer.js [hunk 4]
```diff
@@ -94,10 +96,8 @@ export default function userDashboardReducer(state = defaultState, action) {
         dashboardDialogShown: false,
         userToAdd: defaultState.get("userToAdd"),
         authenticationFailed: false,
-        userPassword: "",
-        newPassword: "",
-        newPasswordConfirmation: "",
-        newPasswordsMatch: true
+        globalPermissions: fromJS({}),
```
Comment (Stezido) [inline frontend/src/pages/Users/reducer.js]: why not use the defaultState here?

File: frontend/src/pages/Workflows/reducer.js [hunk 5]
```diff
@@ -182,6 +185,7 @@ export default function detailviewReducer(state = defaultState, action) {
       return state.merge({
         workflowItemReference: action.wId,
         permissions: fromJS({}),
+        temporaryPermissions: fromJS({}),
```
Comment (Stezido) [inline frontend/src/pages/Workflows/reducer.js]: why not use the defaultState here?

File: frontend/src/pages/Overview/reducer.js (hunk 1 above)
Comment (mathiashoeld) [reply frontend/src/pages/Overview/reducer.js]: done

File: frontend/src/pages/SubProjects/reducer.js (hunk 2 above)
Comment (mathiashoeld) [reply frontend/src/pages/SubProjects/reducer.js]: done

File: frontend/src/pages/Users/Dialog.js (hunk 3 above)
Comment (mathiashoeld) [reply frontend/src/pages/Users/Dialog.js]: done

File: frontend/src/pages/Users/reducer.js (hunk 4 above)
Comment (mathiashoeld) [reply frontend/src/pages/Users/reducer.js]: done

File: frontend/src/pages/Workflows/reducer.js (hunk 5 above)
Comment (mathiashoeld) [reply frontend/src/pages/Workflows/reducer.js]: done

**Pull request merged**